cosmic-core/engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -3239,8 +3226,10 @@ private void advanceStop(final VMInstanceVO vm, final boolean cleanUpEvenIfUnabl
             if (doCleanup) {
                 if (cleanup(vmGuru, new VirtualMachineProfileImpl(vm), work, Event.StopRequested, cleanUpEvenIfUnableToStop)) {
                     try {
-                        if (s_logger.isDebugEnabled()) {
-                            s_logger.debug("Updating work item to Done, id:" + work.getId());
+                        if (work != null) {
+                            if (s_logger.isDebugEnabled()) {
```

Comment:
@borisroman Done. There are 800+ more of these in the code base so still quite some to do